envoy/http/async_client.h

```diff
@@ -324,6 +338,7 @@ class AsyncClient {
     absl::optional<uint32_t> buffer_limit_;
 
     absl::optional<envoy::config::route::v3::RetryPolicy> retry_policy;
+    const Router::RetryPolicy* parsed_retry_policy{nullptr};
```

Comment:
I can only see this used in the test. Is this the only usage? Otherwise this needs to ensure that lifetime of the parsed_retry_policy pointee is longer than the pointer.

Reply:
The intended usage that this feature is being added for is in a filter that's not part of the open source project. And yes the expectation is that the user is passing the AsyncClient a pointer to a RetryPolicy that the AsyncClient doesn't own and which the user needs to ensure will live longer than the AsyncClient. The intended use is to make an AsyncClient call use the RetryPolicy from the selected Route.